Glyssen/Rules/CharacterGroupGenerator.cs

```diff
@@ -914,13 +1021,118 @@ public static void DistributeBooksAmongNarratorGroups(List<AuthorStats> authorSt
 				}
 			}
 
-			private void AssignExtraBiblicalCharactersToExtraBiblicalGroups(List<CharacterDetail> includedCharacterDetails)
+			private void AssignExtraBiblicalCharactersToExtraBiblicalGroups(List<CharacterDetail> includedCharacterDetails,
+				ProjectDramatizationPreferences dramatizationPreferences)
+			{
+				var narratorPrefix = CharacterVerseData.GetCharacterPrefix(CharacterVerseData.StandardCharacter.Narrator);
+
+				// get standard characters not yet assigned
+				foreach (var character in includedCharacterDetails.Where(c => !Groups.Any(g => g.CharacterIds.Contains(c.CharacterId))).ToList())
+				{
+					var characterType = character.StandardCharacterType;
+					switch (characterType)
+					{
+						case CharacterVerseData.StandardCharacter.BookOrChapter:
```

Comment:
I think I would prefer to have each case statement merely set a local variable for the group, followed by a single call to AddExtraBiblicalToGroupOrNarratorGroup using that variable as a parameter, rather than repeating an otherwise-identical method call three times. It would seem to make the the code more compact and reduce the maintenance load. (This would require a default: case with an early return.)